src/improvements/template/SetDisputeGameImpl.sol

```diff
@@ -0,0 +1,111 @@
+import {SuperchainAddressRegistry} from "src/improvements/SuperchainAddressRegistry.sol";
+import {Action} from "src/libraries/MultisigTypes.sol";
+
+import {DisputeGameFactory} from "lib/optimism/packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol";
+import {GameTypes} from "lib/optimism/packages/contracts-bedrock/src/dispute/lib/Types.sol";
+
+import {IFaultDisputeGame} from "lib/optimism/packages/contracts-bedrock/interfaces/dispute/IFaultDisputeGame.sol";
+import {IPermissionedDisputeGame} from
+    "lib/optimism/packages/contracts-bedrock/interfaces/dispute/IPermissionedDisputeGame.sol";
+
+/// @title SetDisputeGameImpl
+/// @notice This template sets the FaultDisputeGame (FDG) and PermissionedDisputeGame (PDG) implementation addresses
+///         in the DisputeGameFactory contract for one or more chains, using values specified in a TOML configuration file.
+///
+/// IMPORTANT: For each chain you wish to update, you MUST provide both FDG and PDG implementation addresses in the TOML file.
+///         - If the provided address matches the current on-chain implementation, the contract will SKIP updating it.
+///         - You are REQUIRED to explicitly specify both addresses for each chain, even if you do not intend to change both.
+///         - To reset an implementation to the zero address, provide "0x000...0" as the value in the TOML.
+///         - Omitting a field is NOT supported and will result in errors and unintended behavior.
+contract SetDisputeGameImpl is L2TaskBase {
+    using stdToml for string;
+
+    /// @notice Struct representing configuration for the task.
+    struct GameImplConfig {
+        uint256 chainId;
+        address fdgImpl;
+        address pdgImpl;
+    }
+
+    /// @notice Mapping of chain ID to configuration for the task.
+    mapping(uint256 => GameImplConfig) public cfg;
+
+    /// @notice Returns the string identifier for the safe executing this transaction.
+    function safeAddressString() public pure override returns (string memory) {
+        return "ProxyAdminOwner";
+    }
+
+    /// @notice Returns string identifiers for addresses that are expected to have their storage written to.
+    function _taskStorageWrites() internal pure virtual override returns (string[] memory) {
+        string[] memory storageWrites = new string[](1);
+        storageWrites[0] = "DisputeGameFactoryProxy";
+        return storageWrites;
+    }
+
+    /// @notice Sets up the template with implementation configurations from a TOML file.
+    function _templateSetup(string memory taskConfigFilePath) internal override {
+        super._templateSetup(taskConfigFilePath);
+        string memory toml = vm.readFile(taskConfigFilePath);
+
+        GameImplConfig[] memory configs = abi.decode(toml.parseRaw(".gameImpls.configs"), (GameImplConfig[]));
+        for (uint256 i = 0; i < configs.length; i++) {
+            cfg[configs[i].chainId] = configs[i];
+        }
```

Comment:
At this point we probably want to add some validation to the FDG addresses. e.g. We shouldn't allow the zero address being set. Also we should do some sanity checking on the address that it's actually a FDG. We should lean on the proofs team here to get their input on what checks we would need.

Reply:
Yeah we should allow the zero address for the FDG when running permissioned proofs. We're actually going to zero these out for Arena Z and Swell based on the proofs team guidance. We'll just add it back when they go permissionless.

Reply:
ok, sounds like we need 👍🏻